tests/Nox.Tests/DatabaseIntegrationTests/SqlServerIntegrationTests.cs

```diff
@@ -165,6 +167,7 @@ public void GeneratedEntity_SqlServer_CanSaveAndReadFields_AllTypes()
         testEntity.StreetAddressTestField!.Value.Should().BeEquivalentTo(addressItem);
         testEntity.AreaTestField!.ToSquareFeet().Should().Be(area);
         testEntity.AreaTestField!.Unit.Should().Be(persistAreaUnitAs);
+        testEntity.VolumeTestField!.ToCubicMeters().Should().Be(volume);
```

Comment:
We need to assert the value with conversion and unit as well. You can check how Area is created and asserted.

Reply:
Added test for Volume Unit.

Reply:
Please create the volume type with specified Units and PersistAs in options, as it was done with Area.

Reply:
Done.